### PR TITLE
Add documentation about using GraphQLQueryWatcher with mutations

### PR DESCRIPTION
**Pull request by damian0815**
Ref https://github.com/apollographql/apollo-ios/issues/3472

### CODE REVIEW

File: apollo-ios/Sources/Apollo/GraphQLQueryWatcher.swift [hunk 1]
```diff
@@ -5,6 +5,8 @@ import ApolloAPI
 
 /// A `GraphQLQueryWatcher` is responsible for watching the store, and calling the result handler with a new result whenever any of the data the previous result depends on changes.
 ///
+/// NOTE: In order for the `resultHandler` to be called when mutations are executed, you will need to implement Custom Cache Keys - see the Caching section of the Apollo iOS documentation for more details.
```
Comment (calvincestari) [inline apollo-ios/Sources/Apollo/GraphQLQueryWatcher.swift:8]: Thanks @damian0815, some suggestions:
* This is not specific, nor limited, to mutations though. If you had another query that used the same fragment you would have the same problem as you experienced before.
* The direction to use custom cache keys is not about having the result handler called but rather about normalizing the response objects that could be shared between operations. The `resultHandler` callback being called is simply the Swift mechanism for getting the result back to your code.

What I'm saying above is that the direction to read about custom cache keys should be about why normalized objects are important for query watchers to function correctly.

**Review by calvincestari — CHANGES_REQUESTED** [review]

File: apollo-ios/Sources/Apollo/GraphQLQueryWatcher.swift (hunk 1 above)
Comment (damian0815) [reply apollo-ios/Sources/Apollo/GraphQLQueryWatcher.swift:8]: okay. i am a newbie to apollo, perhaps you could suggest the exact text and i will update the PR accordingly

Comment (calvincestari) [reply apollo-ios/Sources/Apollo/GraphQLQueryWatcher.swift]: I've pushed two commits with the relevant doc changes.

**Review by calvincestari — APPROVED** [review]

**Comment by calvincestari** [comment]
Going to merge this despite the Docs Preview check failing; I believe that is because it's on a fork. Once merged I'll double-check the published change.

**Comment by damian0815** [comment]
oh @calvincestari I see you merged it already. I'm not sure this is sufficient - the documentation about "normalized cache" in the link you've added is ambiguous about whether Apollo iOS already implements cache normalization. My reading of this document that led to my error with the cache keys not overlapping was, indeed, that Apollo iOS is already normalizing cache keys for me, so there's nothing further for me to do - which is why I was stumped when it didn't work the way I was expecting. 

On closer inspection there's a sentence "if properly configured" in the introduction, but without a link to `SchemaConfiguration.swift` the reader can't really make much use of that statement. Should I open another PR with my suggestion for improvement in that document directly?

**Comment by calvincestari** [comment]
https://github.com/apollographql/apollo-ios-dev/pull/547

**Comment by damian0815** [comment]
thank you that's better yes!

**Pull request merged**